Fix google_generative_ai_conversation tests broken by Python 3.14.3 asyncio changes

### Description

The following `google_generative_ai_conversation` tests are flaky/failing due to the Python 3.14.3 asyncio executor scheduling change (see home-assistant/core#162263 for background).

CPython PR python/cpython#142358 changed how `run_in_executor` delivers results back to the event loop. Results are now set directly on the asyncio future instead of going through `call_soon_threadsafe`, which means `await` returns immediately but other `call_soon_threadsafe` callbacks may not have run yet.

#### Affected tests

**`tests/components/google_generative_ai_conversation/test_init.py::test_devices` (failed 10/10)**

```
tests/components/google_generative_ai_conversation/test_init.py:1323: in test_devices
    assert devices == snapshot
E   AssertionError: assert [+ received] == [- snapshot]
E       list([
E     +   DeviceRegistryEntrySnapshot({
E     +     'area_id': None,
E     +     'config_entries': <ANY>,
E     +     'config_entries_subentries': <ANY>,
E     +     'configuration_url': None,
E     +     'connections': set({...
E
E     ...Full output truncated (58 lines hidden), use '-vv' to show
```

#### What was done

These tests have been temporarily marked with `@pytest.mark.xfail` in PR #169072.

#### What needs to happen

The codeowner should investigate and fix the underlying timing issue in these tests. Common fixes include:
- Adding `await hass.async_block_till_done()` after service calls
- Converting sync entity methods to async
- Restructuring mocks to avoid unintended executor code paths

### Additional context

- Related PR: home-assistant/core#162263
- Disable PR: #169072

--- a/tests/components/google_generative_ai_conversation/snapshots/test_init.ambr
+++ b/tests/components/google_generative_ai_conversation/snapshots/test_init.ambr
@@ -73,15 +73,15 @@
       'identifiers': set({
         tuple(
           'google_generative_ai_conversation',
-          'ulid-ai-task',
+          'ulid-tts',
         ),
       }),
       'labels': set({
       }),
       'manufacturer': 'Google',
-      'model': 'gemini-2.5-flash',
+      'model': 'gemini-2.5-flash-preview-tts',
       'model_id': None,
-      'name': 'Google AI Task',
+      'name': 'Google AI TTS',
       'name_by_user': None,
       'primary_config_entry': <ANY>,
       'serial_number': None,
@@ -102,15 +102,15 @@
       'identifiers': set({
         tuple(
           'google_generative_ai_conversation',
-          'ulid-tts',
+          'ulid-ai-task',
         ),
       }),
       'labels': set({
       }),
       'manufacturer': 'Google',
-      'model': 'gemini-2.5-flash-preview-tts',
+      'model': 'gemini-2.5-flash',
       'model_id': None,
-      'name': 'Google AI TTS',
+      'name': 'Google AI Task',
       'name_by_user': None,
       'primary_config_entry': <ANY>,
       'serial_number': None,

--- a/tests/components/google_generative_ai_conversation/test_init.py
+++ b/tests/components/google_generative_ai_conversation/test_init.py
@@ -1317,8 +1317,9 @@ async def test_devices(
     snapshot: SnapshotAssertion,
 ) -> None:
     """Assert that devices are created correctly."""
-    devices = dr.async_entries_for_config_entry(
-        device_registry, mock_config_entry.entry_id
+    devices = sorted(
+        dr.async_entries_for_config_entry(device_registry, mock_config_entry.entry_id),
+        key=lambda d: d.name,
     )
     assert devices == snapshot
 

